Storage/tests/System/ManageBucketsTest.php

```diff
@@ -163,6 +163,35 @@ public function testUpdateBucketWithLifecycleDeleteRule(array $rule, $isError =
         $this->assertEquals($lifecycle->toArray(), $bucket->info()['lifecycle']);
     }
 
+    public function testUpdateBucketWithAutoclassConfig()
+    {
+        $autoclassConfig = [
+            'autoclass' => [
+                'enabled' => true,
+            ],
+        ];
+
+        $bucket = self::createBucket(
+            self::$client,
+            uniqid(self::TESTING_PREFIX),
+            $autoclassConfig
+        );
+        $this->assertArrayHasKey('autoclass', $bucket->info());
+        $this->assertTrue($bucket->info()['autoclass']['enabled']);
+
+        // test disabling autoclass
+        $autoclassConfig['autoclass']['enable'] = false;
+        $bucket->update($autoclassConfig);
+        $this->assertArrayHasKey('autoclass', $bucket->info());
+        $this->assertFalse($bucket->info()['autoclass']['enabled']);
+
+        // test re-enabling autoclass
+        $autoclassConfig['autoclass']['enable'] = true;
+        $bucket->update($autoclassConfig);
+        $this->assertArrayHasKey('autoclass', $bucket->info());
+        $this->assertTrue($bucket->info()['autoclass']['enabled']);
```

Comment:
#5479 mentions that `once disabled it cannot be enabled again`. So, this test should essentially fail?

Reply:
@saranshdhingra good catch. Lemme confirm with the feature owners. 
@cojenco can you confirm whether `autoclass` can be re-enabled? From our system tests, seems it can be done. I can conclude the same from the feature design doc too, [ref internal link](https://docs.google.com/document/d/1VzJT1fxeaqlZZjqfG9mJJ3YzTOPP_RrjsrYNYb0w9uY/edit?resourcekey=0-WQcYQm_SBKgewPRw7lZIzg#heading=h.uuifumdzb).

Reply:
Autoclass is split into two clickstop launches. For the 1st launch in 10/31/2022 that we're currently working on, customers will only have Autoclass enabled for new buckets only (a bucket can be enabled for Autoclass when created, once disabled it cannot be enabled again).

The reason the tests are passing now is because veneer projects are allowlisted for both new (clickstop 1) and existing (clickstop 2) buckets. See [notes](https://docs.google.com/document/d/11N7-I8wsfSF8a3MAOa2VK__aERo6uMP3ytxe-evwjHM/edit?resourcekey=0-QG15X-e04AMqW-7zjbhcZw#) in internal doc.

I think we can remove testing re-enabling Autoclass for now. It's sufficient to test (1) Autoclass can be enabled upon bucket creation and (2) Autoclass can be disabled upon bucket patch/update. WDYT?

Reply:
@cojenco thanks for the clarification. I have removed the re-enablement test.